contracts/apps/disputable/DisputableApp.sol

```diff
@@ -0,0 +1,186 @@
+        IAgreement agreement = _getAgreement();
+        require(agreement == IAgreement(0) || _agreement == IAgreement(0), ERROR_AGREEMENT_ALREADY_SET);
+
+        AGREEMENT_POSITION.setStorageAddress(address(_agreement));
+        emit AgreementSet(_agreement);
+    }
+
+    /**
+    * @notice Challenge disputable action #`_disputableActionId`
+    * @param _disputableActionId Identification number of the disputable action to be challenged
+    * @param _challengeId Identification number of the challenge in the context of the Agreement
+    * @param _challenger Address challenging the disputable
+    */
+    function onDisputableActionChallenged(uint256 _disputableActionId, uint256 _challengeId, address _challenger) external onlyAgreement {
+        _onDisputableActionChallenged(_disputableActionId, _challengeId, _challenger);
+    }
+
+    /**
+    * @notice Allow disputable action #`_disputableActionId`
+    * @param _disputableActionId Identification number of the disputable action to be allowed
+    */
+    function onDisputableActionAllowed(uint256 _disputableActionId) external onlyAgreement {
+        _onDisputableActionAllowed(_disputableActionId);
+    }
+
+    /**
+    * @notice Reject disputable action #`_disputableActionId`
+    * @param _disputableActionId Identification number of the disputable action to be rejected
+    */
+    function onDisputableActionRejected(uint256 _disputableActionId) external onlyAgreement {
+        _onDisputableActionRejected(_disputableActionId);
+    }
+
+    /**
+    * @notice Void disputable action #`_disputableActionId`
+    * @param _disputableActionId Identification number of the disputable action to be voided
+    */
+    function onDisputableActionVoided(uint256 _disputableActionId) external onlyAgreement {
+        _onDisputableActionVoided(_disputableActionId);
+    }
+
+    /**
+    * @notice Tells whether the Agreement app is a forwarder or not
+    * @dev IForwarder interface conformance
+    * @return Always true
+    */
+    function isForwarder() external pure returns (bool) {
+        return true;
+    }
+
+    /**
+    * @dev Tell the agreement linked to the disputable instance
+    * @return Agreement linked to the disputable instance
+    */
+    function getAgreement() external view returns (IAgreement) {
+        return _getAgreement();
+    }
+
+    /**
+    * @dev Create a new action in the agreement without lifetime set
+    * @param _disputableActionId Identification number of the disputable action in the context of the disputable
+    * @param _submitter Address of the user that has submitted the action
+    * @param _context Link to a human-readable text giving context for the given action
+    * @return Unique identification number for the created action in the context of the agreement
+    */
+    function _newAgreementAction(uint256 _disputableActionId, address _submitter, bytes _context) internal returns (uint256) {
```

Comment:
Not sure if this overload is _that_ useful, and it might be a point of confusion between the two.

Also wonder if you had any strong opinions on the parameter order; `_disputableActionId, _context, _submitter, _lifetime` might feel more natural?

Reply:
I think it avoids the user from calling `_ensureAgreement`, I like the reordering for the parameters :)

Reply:
But the other `_newAgreementAction()` call does the `_ensureAgreement()` check; the user would only have to put the `lifetime` argument in.

Reply:
I think offering different syntax sugar for internal functions is great, especially cause these don't occupy space in terms of bytecode size if you don't use it. However, I know this a personal opinion, we can remove it if it is not aligned with how the rest of the provided implementations in the repo were thought.

Reply:
We've simplified this part now, but on the topic of syntax sugar for internal functions: it will depend, but for the case of a single parameter that defaults to 0, I would hesitate to add it simply because it adds more code to look at and remember.

I wish we had kwargs in solidity, but until that point, I think less can go wrong if you only give someone one way to call something :).